sdk/go/auto/stack_test.go

```diff
@@ -252,3 +254,50 @@ func TestRefreshOptsClearPendingCreates(t *testing.T) {
 
 	assert.Contains(t, args, "--clear-pending-creates")
 }
+
+func TestPreviewImportResources(t *testing.T) {
+	t.Parallel()
+
+	// / Arrange
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping test on Windows")
+	}
+
+	ctx := context.Background()
+	sName := ptesting.RandomStackName()
+	stackName := FullyQualifiedStackName(pulumiOrg, pName, sName)
+
+	s, err := NewStackInlineSource(ctx, stackName, pName, func(ctx *pulumi.Context) error {
+		ctx.Export("exp_static", pulumi.String("foo"))
+		return nil
+	})
+	require.NoError(t, err, "failed to initialize stack")
+
+	defer func() {
+		err = s.Workspace().RemoveStack(ctx, s.Name())
+		assert.Nil(t, err, "failed to remove stack. Resources have leaked.")
+	}()
+
+	tempDir, err := os.MkdirTemp("", "import-files")
+	defer os.RemoveAll(tempDir)
```

Comment:
```suggestion
	tempDir := t.TempDir()
```

This will remove the tempdir automatically when the test is done.